crud/src/main/java/com/redhat/lightblue/mediator/Mediator.java

```diff
@@ -161,12 +164,21 @@ public Response insert(InsertionRequest req) {
                 ctx.getHookManager().callQueuedHooks();
             }
         } catch (Error e) {
+            if(metricCtx!=null) {
+                metricCtx.markRequestException(e);
+            }
             response.getErrors().add(e);
             response.setStatus(OperationStatus.ERROR);
         } catch (Exception e) {
+            if(metricCtx!=null) {
+                metricCtx.markRequestException(e);
```

Comment:
Woops. I neglected to realize that pushing down initializing the metricCtx would create opportunities for failure before we initialized it. That's not good. SO, on second thought, how you originally had it was better :-). My bad. The lesson here is that it's important to balance pushing the metrics collection further down for more information, against pulling the metrics further up to get more context. In this case, my suggestion was a smidgen too far down, missing some potential failure scenarios (we want to get as many of those as possible).